crates/compiler/parser/src/lexer.rs

```diff
@@ -2,20 +2,116 @@
 use logos::Logos;
 use std::fmt;
 
+/// Custom error type for lexing errors in the Cairo-M language.
+///
+/// This enum represents different types of errors that can occur during lexical analysis,
+/// providing detailed information about what went wrong and where.
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub enum LexingError {
+    /// An integer literal that cannot be parsed or is out of range
+    InvalidNumber {
+        /// The problematic number string that failed to parse
+        value: String,
+        /// The reason for the parsing failure
+        reason: NumberParseError,
+    },
+    /// A character that is not recognized as part of any valid token
+    #[default]
+    InvalidCharacter,
+}
+
+/// Specific errors that can occur when parsing numeric literals.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum NumberParseError {
+    /// The number is too large to fit in the target integer type (u32)
+    Overflow,
+    /// The number format is invalid (e.g., invalid characters for the base)
+    InvalidFormat,
+    /// An unknown parsing error occurred
+    Unknown(String),
+}
+
+impl fmt::Display for LexingError {
```

Comment:
I like thiserror. It is also used by SW in stwo. Also those errors do not implement the Error trait unless I miss something?
https://google.github.io/comprehensive-rust/error-handling/thiserror.html